core/rfb.js

```diff
@@ -542,7 +544,17 @@
                     Util.Error("Got data while disconnected");
                     break;
                 case 'normal':
-                    while (this._normal_msg() && this._sock.rQlen() > 0);
+                    while (true) {
+                        if (this._flushing) {
+                            break;
+                        }
+                        if (!this._normal_msg()) {
```

Comment:
> so, here we're replacing a long render queue for a bigger receive queue.

That is the point of the change. The server can detect this and back off. It cannot, however, detect how long our rendering queue is